### PR TITLE
Add page for CF Software list

### PR DESCRIPTION
**Pull request by ethanrd**
An initial stab at adding a CF Software page (see issue #100).

I tried two different ways of formatting the list: a basic Markdown nested list and an HTML collapsable 'details/summary' for each software package. If anyone has a better idea ...

### CODE REVIEW

File: software.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+### cfdm - Python CF Data Model Package
+A description of the [Python CF Data Model Package](https://ncas-cms.github.io/cfdm/)
+
+### cf-python - CF Python Package
+A description of the [CF Python Package](https://ncas-cms.github.io/cf-python/).
+
+### netCDF Flattener - Flatten netCDF files
+The [netCDF Flattener](https://gitlab.eumetsat.int/open-source/netcdf-flattener/) Python package takes netCDF objects that use groups and flattens them while preserving references as described in the [CF Conventions](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#groups).
+The resulting object is logically equivalent to the original, and can be processed by software that isn't able to work with files that use netCDF-4 groups.
+
```
Comment (rabernat) [inline software.md:41]: I would love to include [xarray](http://xarray.pydata.org/) on this list. Xarray is a python package designed for working with labelled, multi-dimensional array data, built around the netCDF data model. It uses CF conventions in several ways, such as encoding / decoding variables and interpreting metadata for visualization.

Comment (rabernat) [reply software.md:41]: Iris of course also needs an entry!

Comment (kevin-obrien) [reply software.md:41]: Should also add PyFerret and ERDDAP to the software list.  I am  happy to provide descriptions.

Comment (ethanrd) [reply software.md:41]: @rabernat - I have grabbed your description above and will include it in my next push to this PR. Do you know who to ping for an Iris description?

@kevin-obrien - Yes, PyFerret and ERDDAP would be good additions.

Comment (rabernat) [reply software.md:74]: Not sure who is maintaining Iris these days, but it's [website](https://scitools.org.uk/iris/docs/latest/) has a pretty concise description.

> Iris implements a data model based on the CF conventions giving you a powerful, format-agnostic interface for working with your data. It excels when working with multi-dimensional Earth Science data, where tabular representations become unwieldy and inefficient.
>
> CF Standard names, units, and coordinate metadata are built into Iris, giving you a rich and expressive interface for maintaining an accurate representation of your data. Its treatment of data and associated metadata as first-class objects includes:

Comment (ethanrd) [reply software.md:86]: Thanks @rabernat. I added Iris as well.

**Comment by ethanrd** [comment]
Clean build. Built page look good. Will merge.

**Pull request merged**